python/ray/worker.py

```diff
@@ -1785,7 +1790,7 @@ def get(object_refs: Union[ray.ObjectRef, List[ray.ObjectRef]],
 @PublicAPI
 @client_mode_hook(auto_init=True)
 def put(value: Any, *,
-        _owner: Optional["ray.actor.ActorHandle"] = None) -> ray.ObjectRef:
+        _owner: Optional["ray.actor.ActorHandle"] = None, weakref: bool = False) -> ray.ObjectRef:
```

Comment:
this will allow ray to GC the objects that were put as a weakref -- we use it for the checkpoints that we send to workers for loading, given that once this action has been done it's OK to drop them (technically, dropping the references in pythonland should allow ray to collect form the object store, but we've obviously had trouble with that in the past)

Reply:
(out of scope for this PR) Given that this is not officially supported by `ray` can we find another workaround/search for object id leakage